homeassistant/components/heatmiser/climate.py

```diff
@@ -41,36 +39,34 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the heatmiser thermostat."""
     from heatmiserV3 import heatmiser, connection
 
-    ipaddress = config.get(CONF_IPADDRESS)
-    port = str(config.get(CONF_PORT))
-    tstats = config.get(CONF_TSTATS)
+    HT = heatmiser.HeatmiserThermostat
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    thermostats = config.get(CONF_THERMOSTATS)
 
-    serport = connection.connection(ipaddress, port)
-    serport.open()
+    UH1 = connection.HeatmiserUH1(host, port)
```

Comment:
I assume that there will be a traceback if the unit is not found.

Also, variable name.

Reply:
Fixed variable name, there is a timeout if a wrong address is entered, within a couple of seconds.  I attempted to rewrite this in the config_flow framework, but failed.